testsuite/src/test/java/org/jboss/intersmash/tests/wildfly/message/broker/amq/WildflyJmsSslApplication.java

```diff
@@ -0,0 +1,169 @@
+*
+*         http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+package org.jboss.intersmash.tests.wildfly.message.broker.amq;
+
+import cz.xtf.builder.builders.pod.PersistentVolumeClaim;
+import cz.xtf.builder.builders.pod.VolumeMount;
+import io.fabric8.kubernetes.api.model.EnvVar;
+import io.fabric8.kubernetes.api.model.EnvVarBuilder;
+import io.fabric8.kubernetes.api.model.Secret;
+import io.fabric8.kubernetes.api.model.SecretBuilder;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Base64;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import org.apache.commons.io.FileUtils;
+import org.assertj.core.util.Strings;
+import org.jboss.intersmash.IntersmashConfig;
+import org.jboss.intersmash.application.input.BuildInput;
+import org.jboss.intersmash.application.input.BuildInputBuilder;
+import org.jboss.intersmash.application.openshift.WildflyImageOpenShiftApplication;
+import org.jboss.intersmash.tests.wildfly.WildflyApplicationConfiguration;
+import org.jboss.intersmash.tests.wildfly.util.SimpleCommandLineBasedKeystoreGenerator;
+
+public class WildflyJmsSslApplication implements WildflyImageOpenShiftApplication, WildflyApplicationConfiguration {
+	static final String NAME = "wildfly-jms";
+	protected final List<Secret> secrets = new ArrayList<>();
+	public static final String WILDFLY_TRUSTSTORE_SECRET_NAME = String.format("%s-truststore-secret", NAME);
+	private final BuildInput buildInput;
+	private final Map<PersistentVolumeClaim, Set<VolumeMount>> persistentVolumeClaimMounts;
+	private final List<EnvVar> environmentVariables;
+	private static final String WILDFLY_TRUSTSTORE_DIR_NAME = "/etc/secrets";
+	private static final String WILDFLY_TRUSTSTORE_FILE_NAME = "client.ts";
+
+	public WildflyJmsSslApplication() throws IOException {
+		final String applicationDir = "wildfly/activemq-artemis-ssl";
+		buildInput = new BuildInputBuilder()
+				.uri(IntersmashConfig.deploymentsRepositoryUrl())
+				.ref(IntersmashConfig.deploymentsRepositoryRef())
+				.build();
+
+		final SimpleCommandLineBasedKeystoreGenerator.CertificateInfo infinispanCertificate = SimpleCommandLineBasedKeystoreGenerator
+				.generateCertificate(
+						AmqBrokerSslApplication.getAcceptorServiceName(),
+						AmqBrokerSslApplication.KEYALIAS,
+						AmqBrokerSslApplication.STOREPASS,
+						Collections.emptyList());
+		/*
+		    apiVersion: v1
+		    kind: Secret
+		    metadata:
+		      name: ex-aao-truststore-secret
+		    type: Opaque
+		    stringData:
+		      alias: server
+		      trustStorePassword: 1234PIPPOBAUDO
```

Comment:
And there's another one... This thing will end up in a book that I'll write when I am old.